packages/eslint-plugin/docs/rules/no-unnecessary-type-parameters.md

```diff
@@ -0,0 +1,104 @@
+---
+description: 'Disallow type parameters that only appear once'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unnecessary-type-parameters** for documentation.
+
+This rule forbids type parameters that only appear once in a function, method, or class definition.
+
+Type parameters relate two types. If a generic type parameter only appears once, then it is not relating anything. It can usually be replaced with explicit types such as `unknown`.
+
+At best unnecessary type parameters make code harder to read. At worst they can be used to disguise unsafe type assertions.
+
+## Options
+
+None.
+
+## Examples
+
+<!--tabs-->
+
+### ❌ Incorrect
+
+```ts
+function third<A, B, C>(a: A, b: B, c: C): C {
```

Comment:
[Docs] Nit: `third` is the hardest to parse out of the examples for me, while `parseYAML` is a pretty much perfect showcase of the rule. What do you think about making `third` either last or second-to-last, and or making the name slightly more descriptive - like `onlyThirdTypeParameterIsNeeded`?